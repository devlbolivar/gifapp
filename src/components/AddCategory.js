import React,{useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {

        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories(categories => [ inputValue,...categories]);
            setInputValue('');
        }

    }


    return (
        <form className="form__search__bar" onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                placeholder="Add category"
                onChange={handleInputChange}
            />
            <button>ok</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}