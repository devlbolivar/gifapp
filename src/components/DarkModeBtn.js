import React from 'react'

export const DarkModeBtn = () => {
    return (
        <div className="push">
            <span>bot√≥n</span>
        </div>
    );
}
