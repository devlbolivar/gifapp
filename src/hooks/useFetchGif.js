import { useState, useEffect } from 'react';
import { getGif } from '../helpers/getGif';

export const useFetchGif = () => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(()=>{
        getGif()
        .then(img => {
            setState({ 
                data: img,
                loading: false
            });
        });
    }, []);

    return state; 
}