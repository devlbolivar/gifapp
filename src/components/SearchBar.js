import React from 'react';
import { AddCategory } from './AddCategory';

export const SearchBar = ({setCategories}) => {
    return (
        <div>
            <AddCategory setCategories={setCategories} />
        </div>
    )
}
