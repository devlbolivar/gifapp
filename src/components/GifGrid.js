//import React, { useState, useEffect } from 'react'
import React from 'react'
import { GifGridItem } from './GifGridItem';
import {useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
    
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">loading</p> }
            <div className="card-grid">
                { 
                    images.map(image =>(
                        <GifGridItem 
                            key={image.id} 
                            {...image} 
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
