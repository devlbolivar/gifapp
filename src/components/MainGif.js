import React from "react";
import { useFetchGif } from "../hooks/useFetchGif";

export const MainGif = () => {
  useFetchGif();
  return (
    <div className="main__gif">
      <img alt="Random Img logo" />
    </div>
  );
};
