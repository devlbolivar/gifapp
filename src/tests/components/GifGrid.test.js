import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba sobre el componente <GifGrid />', ()=>{
    
    const category = 'funny';
    
    test('Debe renderizar el componente correctamente', ()=>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid key={category} category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    // test('Debe de mostrar items cuando se cargan imÃ¡genes con el useFetchGifs', ()=>{

    //     const gifs = [{
    //         it: 'abc',
    //         url: 'https://prueba.com',
    //         title: 'Prueba'
    //     }]
    //     useFetchGifs.mockReturnValue({
    //         data: gifs,
    //         loading: false
    //     });

    //     const wrapper = shallow(<GifGrid key={category} category={category} />);

    //     expect(wrapper).toMatchSnapshot();

    // });
});