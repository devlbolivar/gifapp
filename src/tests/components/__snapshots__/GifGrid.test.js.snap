// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el componente <GifGrid /> Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    funny
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
